src/charm.py

```diff
@@ -112,41 +112,40 @@ def __init__(self, *args):
         self.framework.observe(self._common_database.on.endpoints_changed, self._configure_webui)
         self.framework.observe(self._auth_database.on.endpoints_changed, self._configure_webui)
         self.framework.observe(
-            self.on.sdcore_management_relation_joined, self._publish_sdcore_management_url
+            self.on.sdcore_management_relation_joined, self._configure_webui
         )
         # Handling config changed event to publish the new url if the unit reboots and gets new IP
-        self.framework.observe(self.on.config_changed, self._publish_sdcore_management_url)
+        self.framework.observe(self.on.config_changed, self._configure_webui)
 
-    def _configure_webui(self, event: EventBase) -> None:
-        """Handle config changes.
+    def _configure_webui(self, _: EventBase) -> None:
+        """Configure Webui configuration file and publishes the Webui management URL.
 
-        Manage pebble layer and Juju unit status.
+        The main callback method for `config changed`, `pebble ready`, `relation changed` e.g.
+         and custom events. Manages the Pebble services.
```

Comment:
```suggestion
        Whenever a Juju event is emitted, this method is called to configure the Webui workload, run it and expose the service information through charm's interface. Prior to running any of the above, the method performs a couple of checks to make sure that the workload is ready to be started. 
```
How 'bout something along these lines?

Reply:
Done